python/lsst/meas/base/forcedMeasurement.py

```diff
@@ -368,12 +350,21 @@ def generateSources(self, exposure, refCat, refWcs, idFactory=None):
         for ref in refCat:
             newSource = sources.addNew()
             newSource.assign(ref, self.mapper)
-            footprint = newSource.getFootprint()
-            if footprint is not None:
-                # if heavy, just transform the "light footprint" and leave the rest behind
-                if footprint.isHeavy():
-                    footprint = lsst.afw.detection.Footprint(footprint)
-                if not refWcs == targetWcs:
-                    footprint = footprint.transform(refWcs, targetWcs, expRegion, True)
-                newSource.setFootprint(footprint)
         return sources
+
+    def attachTransformedFootprints(self, sources, refCat, exposure, refWcs):
+        """!Default implementation for attaching Footprints to blank sources prior to measurement
+
+        Footprints for forced photometry must be in the pixel coordinate system of the image being
+        measured, while the actual detections may start out in a different coordinate system.
+        This default implementation transforms the Footprints from the reference catalog from the
+        refWcs to the exposure's Wcs, which downgrades HeavyFootprints into regular Footprints,
+        destroying deblend information.
+
+        Note that ForcedPhotImageTask inherits this method and all its derived classes must
+        implement this method to define how their Footprints should be generated.
```

Comment:
It's not actually "inherits": `ForcedPhotImageTask` _delegates_ to this method in its own `attachFootprints` method, so it's subclasses will use this implementation unless they override.